### PR TITLE
Upgrade to Ionic 7

### LINKED ISSUE #3047: [BUG] toggle should not emit change event when checked is programatically set
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
kirby-toggle emits checkedChange its `checkedChange` event whenever the checked-input. 
This is not the expected behavior, and can cause logic to run that should only be activated on user input.

In this example with the angular material toggle nothing is logged to console before you manually toggle it: https://stackblitz.com/edit/mqhnz6?file=src%2Fapp%2Fslide-toggle-overview-example.html 
 
## Describe how to reproduce the bug

1. Go to https://cookbook.kirby.design/#/home/showcase/toggle
2. Inspect the first toggle and highlight it in the Elements tab in the inspector tab
3. Go to the console tab and input `$0.checked = true`
4. See that the checkedChange event from the example is fired, which it should only do when pressed by a user

## Which Kirby version was used?
8.4.2

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/main/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

### LINKED ISSUE #3246: [Housekeeping] Upgrade to Ionic 7
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the housekeeping request
Ionics maintainance [status page](https://ionicframework.com/docs/reference/support#framework-maintenance-and-support-status) shows that support for Ionic v6 ends September 19, 2023, so we should update to v7. 

## Are there any additional context?
We have done a preliminary test of Ionic v7 and found the following issues to be fixed:
- [x] Checkboxes have wrong background color and border color
- [x] ticks on the range component are misaligned
- [x] reorder-list items are taller than they should be (turns out to be toggle in list that gets wrong padding)
- [x] radio in item has wrong dimension
- [x] segmented control default mode does not fit segment width to text
- [x] segmented control chip and compact chip stretch to full width
- [x] label not clickable on radio
- [x] range disabled state wrong color

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #3246 closes #3047

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts [hunk 1]
```diff
@@ -17,7 +16,7 @@ export class ModalAnimationBuilderService {
   };
 
   public enterAnimation(currentBackdrop?: HTMLIonBackdropElement): AnimationBuilder {
-    return (baseEl: HTMLElement, opts: ModalAnimationOptions): Animation => {
+    return (baseEl: HTMLElement, opts: any): Animation => {
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts]: This looks funky, but between v6.7.5 and v7.3.3, the core package exports was changed from default exports to named exports, and now `ModalAnimationOptions` is only available internally to Ionic. 

v6.7.5:
`export * from './components/modal/modal-interface';`

v7.3.3:
`export { ModalOptions, ModalCustomEvent } from './components/modal/modal-interface';`

Comment (RasmusKjeldgaard) [reply libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts]: So, should we copy out the type from Ionic core, or continue with `any`?

File: libs/designsystem/checkbox/src/checkbox.component.ts [hunk 2]
```diff
@@ -28,7 +35,7 @@ export class CheckboxComponent {
 
   @HostBinding('class')
   @Input()
-  size?: 'xs' | 'sm' | 'md';
+  size?: 'xs' | 'sm' | 'md' = 'md';
```
Comment (jakobe) [inline libs/designsystem/checkbox/src/checkbox.component.ts]: Nitty gritty, but I guess this is no longer optional (i.e. can be `undefined`) if it has a default value? 🤔 
```suggestion
  size: 'xs' | 'sm' | 'md' = 'md';
```

File: libs/designsystem/helpers/src/ionic-element-part-service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { ElementRef, Injectable, Renderer2 } from '@angular/core';
+
+@Injectable()
+export class IonicElementPartService {
```
Comment (jakobe) [inline libs/designsystem/helpers/src/ionic-element-part-service.ts]: Consider naming - all other helpers in this folder ends with `-helper`...

File: libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts (hunk 1 above)
Comment (jakobe) [reply libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts]: @RasmusKjeldgaard maybe just inline a type with the property from `ModalAnimationOptions` that we need?
```suggestion
    return (baseEl: HTMLElement, opts: { presentingEl?: HTMLElement }): Animation => {
```

File: libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts [hunk 4]
```diff
@@ -141,11 +140,7 @@ export class ModalAnimationBuilderService {
   }
 
   public leaveAnimation(currentBackdrop?: HTMLIonBackdropElement): AnimationBuilder {
-    return (
-      baseEl: HTMLElement,
-      opts: ModalAnimationOptions,
-      duration = this.duration
-    ): Animation => {
+    return (baseEl: HTMLElement, opts: any, duration = this.duration): Animation => {
```
Comment (jakobe) [inline libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts]:
```suggestion
    return (baseEl: HTMLElement, opts: { presentingEl?: HTMLElement }, duration = this.duration): Animation => {
```

File: libs/designsystem/page/src/page.component.ts [hunk 5]
```diff
@@ -388,7 +390,11 @@ export class PageComponent
 
   ngOnInit(): void {
     this.removeWrapper();
-    this.setToolbarBackgroundPart();
+    this.ionicElementPartService.setPart(
+      'background',
+      this.ionToolbarElement,
+      '.toolbar-background'
+    );
```
Comment (jakobe) [inline libs/designsystem/page/src/page.component.ts]: Nice! 👌🏻

File: libs/designsystem/radio/src/radio.component.ts [hunk 6]
```diff
@@ -25,7 +30,7 @@ export class RadioComponent {
 
   @HostBinding('class')
   @Input()
-  size?: 'xs' | 'sm' | 'md';
+  size?: 'xs' | 'sm' | 'md' = 'md';
```
Comment (jakobe) [inline libs/designsystem/radio/src/radio.component.ts]: Nitty gritty, but I guess this is no longer optional (i.e. can be `undefined`) if it has a default value? 🤔 
```suggestion
  size: 'xs' | 'sm' | 'md' = 'md';
```

**Review by jakobe — CHANGES_REQUESTED** [review]
A few small things—otherwise great work! 💪🏻 👌🏻 👏🏻

File: libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts (hunk 1 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/modal/src/modal/services/modal-animation-builder.service.ts]: Excellent, good idea!

File: libs/designsystem/helpers/src/ionic-element-part-service.ts (hunk 3 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/helpers/src/ionic-element-part-service.ts]: Renamed to `IonicElementPartHelper` in 7b2134cc6805ff04c46a02cc0cc75193ba61b052

**Review by jakobe — APPROVED** [review]

**Pull request merged**